nixos/modules/services/misc/freeswitch.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ...}:
+with lib;
+let
+  cfg = config.services.freeswitch;
+  pkg = cfg.package;
+  configDirectory = pkgs.runCommand "freeswitch-config-d" { } ''
+    mkdir -p $out
+    cp -r ${pkg}/etc/freeswitch/* $out
+    chmod -R +w $out
+    ${concatStringsSep "\n" (mapAttrsToList (file: text: ''
+      mkdir -p $out/$(dirname ${file})
+      cat >$out/${file} <<'EOF'
+      ${text}
+      EOF
+    '') cfg.configDir)}
+  '';
+  configPath = if cfg.enableReload
+    then "/etc/freeswitch"
+    else configDirectory;
+in {
+  options = {
+    services.freeswitch = {
+      enable = mkEnableOption "FreeSWITCH";
+      enableReload = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Reload FreeSWITCH when configuration directory changes (instead of restart).
```

Comment:
Reload will call ```fs_cli -x reloadxml``` now, ```SIGHUP``` is not properly documented and seem to only reopen log files. ```reloadxml``` might not be what user want so better to have this disabled by default.

Reply:
What does not get reloaded with just `reloadxml`? Is there something that needs a restart for a config directory change?

Reply:
I have no idea but this is the first line from [docs](https://freeswitch.org/confluence/display/FREESWITCH/Reloading):
>When is a ```reloadxml``` sufficient for "reloading and moving on"? This can be a tricky question. Sometimes reloading XML is sufficient. Sometimes more is needed.